### DISCUSSION TITLE
problem with BulkAction

### DISCUSSION THREAD

**Discussion by newtec-foggia** [filamentphp/filament#10642] [Help]
Hi,
I have a user table with 50,000 records, I noticed when creating a bulk action that when trying to select all the records it doesn't start any jobs or rather it seems to be blocked, for smaller selections it works perfectly. It seems to block the processing of the Collection, in fact the request to confirm processing does not even appear...
![code ](https://github.com/filamentphp/filament/assets/83398776/98fa48b2-8e01-4aba-b5cb-69a887179da8)

**Comment by demobiel** [comment]
I think this is a memory issue. 

I wouldn't use the $records, but rather the $livewire->selectedTableRecords, which will return you an array the ID's of the selected records. And work with this array, instead of the records collection. Pass the chunk of ID's to your job, and only get the records once inside the job.

But I might be wrong, but this is how I work now with bulkactions.

**Comment by newtec-foggia** [comment]
Thank you, I followed your instructions, it speeded up but it still fails to process the queue for many records (50k)..can you work out these numbers?
![code ](https://github.com/filamentphp/filament/assets/83398776/ab1c663d-0c78-47f1-b2ff-3225f23e2cac)

**Comment by demobiel** [comment]
I would create one job that accepts the array of 50.000 IDs, and creates separate jobs. So don't do the dispatch all the jobs in the Filament Action, it just takes too much time, and I think you are facing a time out.

I would dispatch one job (ExportUsersJob), give it the ID's and let that job handle the dispatching of the other jobs (WriteChunkToCSV).

**Reply by newtec-foggia** [reply]
I'm sorry but it's not clear to me, the problem I'm having is precisely in the transition to the main job and then being able to process in blocks... can you give me a more detailed example?
![code ](https://github.com/filamentphp/filament/assets/83398776/bc9ecd8c-bba5-46cc-9420-6cf7c49956e7)

**Comment by demobiel** [comment]
well what you are doing now, is exactly what I would do, and in the main job dispatch the child jobs to actually write to the csv.

Can you describe what is happening now? Is the main job never dispatched? Is the Notification shown?

**Reply by newtec-foggia** [reply]
after pressing the batch export button, the waiting time is too much... and we are talking about 10k records...
https://github.com/filamentphp/filament/assets/83398776/d68d7ff9-d927-4991-a24b-8157602c7ab2

**Comment by demobiel** [comment]
How your queue connection defined in your .env file? from what I see this is still in sync, which means the job is still executed in sync.

**Reply by newtec-foggia** [reply]
no database
![code ](https://github.com/filamentphp/filament/assets/83398776/74d5e950-4262-40be-8d8f-1f23b82fba59)

**Comment by demobiel** [comment]
and if I understand well the job is never created in the database? does it work with smaller selection? is there anything logged in laravel.log file?

**Reply by newtec-foggia** [reply]
the job is always created, the problem is as you can see from the video, with the increase in the selected records the time that passes between pressing the button and the request to confirm the operation is too long

**Reply by demobiel** [reply]
Yes I see now... so actually the job has nothing to do with it? Even an "empty" action reproduces this? Like just dd($livewire->selectedTableRecords) ?

**Reply by newtec-foggia** [reply]
yes, in fact, even an empty action... it seems to waste a lot of time after pressing the export button

**Reply by demobiel** [reply]
aha, and if you remove the $livewire from the closure and just a simple dd("action called"); ? I start to have the feeling that there is something going on when Filament tries to resolve the arguments of the closure.

**Reply by newtec-foggia** [reply]
the problem seems to be the action.. because with selection of a lower number of records it's fine.. I don't have waiting times.. but I repeat when I press the export button..

![code ](https://github.com/filamentphp/filament/assets/83398776/c589e413-7cce-4e05-b56b-80c0ab747781)

**Reply by demobiel** [reply]
my last clue is: what if you don't require confirmation? are any of the other functions causing delays?

Otherwise, I have no clue, and I would raise an issue...

**Reply by newtec-foggia** [reply]
same problem even deleting the confirmation.. Can you do the same test with a table of at least 10/15k records?

**Reply by demobiel** [reply]
in that case I am really sorry, but I can't help you. We need some experts on this.

I have not a project right now where I can do this test... I have this principle where we had up to 5000 records, and there is some delay, but not as much as yours...

**Reply by newtec-foggia** [reply]
yes no problem with 5000 records.. ok let's see if there are other suggestions otherwise I'll open a ticket.. thank you very much

**Reply by demobiel** [reply]
I would look at the file Filament\Tables\Actions\BulkAction and there is a call function... and start from there in my debugging...

<img width="731" alt="image" src="https://github.com/filamentphp/filament/assets/390530/004aaa75-bfee-441e-8b39-999be444a0d2">

**Reply by newtec-foggia** [reply]
do you suggest reporting the probable bug?

**Reply by demobiel** [reply]
the best way is report is, is to create an empty filament project with a seeder of 10000 records. And see if the problem is still there. If so, report as a bug or enhancement, and point to that repository. This helps the maintainers to quickly see what is going on, and work on a solution.

**Reply by newtec-foggia** [reply]
I opened the report
